### DISCUSSION TITLE
Edge functions timeout

### DISCUSSION THREAD

**Discussion by andreyDimitrov** [supabase/supabase#14325] [Questions]
I have edge functions implemented to request completions from OpenAI. With the switch to GPT-4, those actually are taking more than a minute to come back. So my client gets a lovely `ERR_FAILED 504 (Gateway Time-out)` at exactly 1 minute. My edge function itself keeps running and I see the OpenAI response in the logs (at around 90 sec mark). How can I extend the timeout on the client when waiting on the result of the edge function?

**Comment by jasonklein** [comment]
I have the same question. Did you get an answer or find a workaround, @andreyDimitrov?

**Comment by jziggas** [comment]
I have also just learned this the hard way while working on something at my day job. Now I'm not even sure how my team is supposed to proceed with this feature since the wall clock timeout is supposedly hard coded by Deno Deploy itself and we are not interested in self-hosting Edge Functions.
